pkg/primitive/return.go

```diff
@@ -0,0 +1,93 @@
+package primitive
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hyperifyio/gnd/pkg/log"
+)
+
+// Return represents the return primitive
+type Return struct {
+	// IsSubroutine indicates whether this primitive is being executed in a subroutine context
+	IsSubroutine bool
```

Comment:
Not sure why we need to know that it's a subroutine. The executing context (eg. script or subroutine) should handle logic related to that.